HACKING.md

```diff
@@ -0,0 +1,90 @@
+# Hacking on Libreport
+
+Here's where to get the code:
+
+    $ git clone https://github.com/abrt/libreport.git
+    $ cd libreport/
+
+The remainder of the commands assume you're in the top level of the
+Libreport git repository checkout.
+
+## Building
+
+1. Install dependencies
+
+    First you should install all dependent packages.
+
+    Dependencies can be listed by:
+
+        $ ./autogen.sh sysdeps
+
+    or installed by:
+
+        $ ./autogen.sh sysdeps --install
+
+    The dependency installer gets the data from [the rpm spec file](libreport.spec.in)
+
+2. Build from source
+
+    When you have all dependencies installed you can now build a rpm package by these commands:
+
+        $ ./autogen.sh
+        $ make rpm
+
+    Now in the `build` folder you can find a rpm packages both in `noarch` and `x86_64` (or your
+            architecture) directories.
+
+
+### Testing
+
+Easiest way how to test everything (build, run tests) is to make rpm build (see,
+        *2. Build from source in Building chapter*).
+
+For running only tests execute:
+
+    ./autogen.sh
+    make check
+
+(Note: If you put any arguments to `./autogen.sh` command, put also `--prefix=/usr` for all tests to
+ work correctly)
+
+If you need to only rerun one specific test:
+
+    cd tests
+    AUGEAS_LENS_LIB="/usr/share/augeas/lenses:<clone_path>/augeas" ./testsuite <n>
+
+where \<clone\_path\> is path to your top level of the Libreport git repository checkout and \<n\> is
+number of the test.
+
+If you make changes in a test, before running it again, execute:
+
+    make  atconfig atlocal ./testsuite
+
+## Contributing a change
+
+### Basic git workflow:
+
+1. Fork the Libreport repository (hit fork button on https://github.com/abrt/libreport)
+
+2. Clone your fork
+
+3. Checkout to a new branch in your clone (`git checkout -b <name_of_branch>`)
```

Comment:
“Checkout” is technically a noun. :p 

“Switch to a new branch”?

Reply:
```suggestion
3. Create and check out a new branch in your clone (`git checkout -b <name_of_branch>`)
```

Reply:
Couldn't refuse to try out the new toy "Insert a suggestion".